### PR TITLE
Bump up go and runc for 1.31

### PR DESCRIPTION
**Pull request by Issacwww**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
address vulnerabilities
https://pkg.go.dev/vuln/GO-2024-3110
https://pkg.go.dev/vuln/GO-2024-3107

### CODE REVIEW

File: .go-version [hunk 1]
```diff
@@ -1 +1 @@
-1.22.6
```
Comment (dims) [inline .go-version:1]: We did this 3 weeks ago... https://github.com/kubernetes/kubernetes/commit/a732fb11673778368ff69dfe37092382ecc1f311

Comment (dims) [reply .go-version:1]: in general, updating go versions is a long process FYI - https://github.com/kubernetes/release/issues/3738

**Comment by dims** [comment]
please drop the golang update in this PR and keep it focused on https://github.com/opencontainers/runc/security/advisories/GHSA-jfvp-7x6p-h2pv

Also, A quick glance at the patches for runc, i am not sure if kubelet itself is affected, but worth checking with the sig-node tech leads.

**Comment by Issacwww** [comment]
the runc one seems not impacting kubelet, but local scan get other vulnerabilities reported, is there any ongoing issue addressing them?
```
govulncheck -mode=binary _output/bin/kubelet
=== Symbol Results ===

Vulnerability #1: GO-2024-2748
    Privilege Escalation in Kubernetes in k8s.io/apimachinery
  More info: https://pkg.go.dev/vuln/GO-2024-2748
  Module: k8s.io/apimachinery
    Found in: k8s.io/apimachinery@(devel)
    Fixed in: k8s.io/apimachinery@v0.16.13
    Vulnerable symbols found:
      #1: net.ConnectWithRedirects
      #2: proxy.UpgradeAwareHandler.ServeHTTP

Vulnerability #2: GO-2023-2113
    Memory exhaustion in go.opentelemetry.io/contrib/instrumentation
  More info: https://pkg.go.dev/vuln/GO-2023-2113
  Module: go.opentelemetry.io/contrib/instrumentation/github.com/emicklei/go-restful/otelrestful
    Found in: go.opentelemetry.io/contrib/instrumentation/github.com/emicklei/go-restful/otelrestful@v0.42.0
    Fixed in: go.opentelemetry.io/contrib/instrumentation/github.com/emicklei/go-restful/otelrestful@v0.44.0
    Vulnerable symbols found:
      #1: semconvutil.HTTPClientRequest
      #2: semconvutil.HTTPServerRequest
      #3: semconvutil.httpConv.ClientRequest
      #4: semconvutil.httpConv.ServerRequest

Vulnerability #3: GO-2022-0965
    Unbounded recursion in JSON parsing in k8s.io/apimachinery
  More info: https://pkg.go.dev/vuln/GO-2022-0965
  Module: k8s.io/apimachinery
    Found in: k8s.io/apimachinery@(devel)
    Fixed in: k8s.io/apimachinery@v0.0.0-20190927203648-9ce6eca90e73
    Vulnerable symbols found:
      #1: json.Serializer.Decode
      #2: json.Serializer.Encode
      #3: json.Unmarshal
      #4: json.customNumberDecoder.Decode

Vulnerability #4: GO-2021-0065
    Unauthorized credential disclosure in k8s.io/kubernetes and k8s.io/client-go
  More info: https://pkg.go.dev/vuln/GO-2021-0065
  Module: k8s.io/client-go
    Found in: k8s.io/client-go@(devel)
    Fixed in: k8s.io/client-go@v0.17.0
    Vulnerable symbols found:
      #1: transport.basicAuthRoundTripper.RoundTrip
      #2: transport.basicAuthRoundTripper.RoundTrip
      #3: transport.bearerAuthRoundTripper.RoundTrip
      #4: transport.bearerAuthRoundTripper.RoundTrip
      #5: transport.debuggingRoundTripper.RoundTrip
      Use '-show traces' to see the other 5 found symbols

Vulnerability #5: GO-2021-0064
    Unauthorized credential disclosure via debug logs in k8s.io/kubernetes and
    k8s.io/client-go
  More info: https://pkg.go.dev/vuln/GO-2021-0064
  Module: k8s.io/client-go
    Found in: k8s.io/client-go@(devel)
    Fixed in: k8s.io/client-go@v0.20.0-alpha.2
    Vulnerable symbols found:
      #1: transport.basicAuthRoundTripper.RoundTrip
      #2: transport.basicAuthRoundTripper.RoundTrip
      #3: transport.bearerAuthRoundTripper.RoundTrip
      #4: transport.bearerAuthRoundTripper.RoundTrip
      #5: transport.debuggingRoundTripper.RoundTrip
      Use '-show traces' to see the other 5 found symbols
```

**Pull request closed without merging**